supabase/functions/oauth/encrypt-config.ts

```diff
@@ -3,64 +3,76 @@ import { createClient } from "https://esm.sh/@supabase/supabase-js";
 import Handlebars from "https://esm.sh/handlebars";
 import jsonpointer from "https://esm.sh/jsonpointer.js";
 import { corsHeaders } from "../_shared/cors.ts";
+import { returnPostgresError } from "../_shared/helpers.ts";
+import { supabaseClient } from "../_shared/supabaseClient.ts";
 
 const ENCRYPTION_SERVICE =
   "https://config-encryption.estuary.dev/v1/encrypt-config";
 
 const CREDENTIALS_KEY = "credentials";
 
 export async function encryptConfig(req: Record<string, any>) {
-  const supabase = createClient(
-    Deno.env.get("SUPABASE_URL")!,
-    Deno.env.get("SUPABASE_SERVICE_ROLE_KEY")!
-  );
+  const { connector_id, connector_tag_id, config } = req;
 
-  const { connector_id, config, schema } = req;
-
-  const { data, error } = await supabase
+  const { data, error } = await supabaseClient
     .from("connectors")
     .select("oauth2_client_id,oauth2_client_secret")
     .eq("id", connector_id)
     .single();
 
   if (error != null) {
-    return new Response(JSON.stringify({ error }), {
-      headers: {
-        ...corsHeaders,
-        "Content-Type": "application/json",
-      },
-      status: 400,
-    });
+    returnPostgresError(error);
   }
 
   const { oauth2_client_id, oauth2_client_secret } = data;
 
-  config[CREDENTIALS_KEY]["client_id"] = oauth2_client_id;
-  config[CREDENTIALS_KEY]["client_secret"] = oauth2_client_secret;
+  if (config[CREDENTIALS_KEY]) {
+    config[CREDENTIALS_KEY]["client_id"] = oauth2_client_id;
+    config[CREDENTIALS_KEY]["client_secret"] = oauth2_client_secret;
+  }
+
+  const { data: connectorTagData, error: connectorTagError } =
+    await supabaseClient
+      .from("connector_tags")
+      .select("endpoint_spec_schema")
+      .eq("id", connector_tag_id)
+      .single();
+
+  if (connectorTagError != null) {
+    returnPostgresError(error);
+  }
```

Comment:
Here we are now going to fetch the connector tag schema instead of having the UI send it. This way no one can mess with it causing issues for the encryption service.